src/commonMain/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -93,7 +93,15 @@ internal class YamlOutput(
         }
     }
 
-    override fun encodeEnum(enumDescriptor: SerialDescriptor, index: Int) = emitQuotedScalar(enumDescriptor.getElementName(index), configuration.singleLineStringStyle.scalarStyle)
+    override fun encodeEnum(enumDescriptor: SerialDescriptor, index: Int) {
+        val element = if (configuration.decodeEnumCaseInsensitive) { // use lower case enums
+            enumDescriptor.getElementName(index).lowercase()
```

Comment:
note: The configuration property states that it is related to decoding. Probably, the encoding should not be changed at all

Reply:
This is done to make the encoding consistent with how json handles it.

Reply:
By "json", do you mean kotlinx-serialization-json library?

Reply:
Yes

Reply:
I might be missing something, but I don't see a conversion to lower-case in kotlinx-serialization-json
- [stream encoding](https://github.com/Kotlin/kotlinx.serialization/blob/master/formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonEncoder.kt#L228)
- [tree encoding](https://github.com/Kotlin/kotlinx.serialization/blob/master/formats/json/commonMain/src/kotlinx/serialization/json/internal/TreeJsonEncoder.kt#L104)

Could you please point out where this is handled?

Reply:
Sorry, I misinterpreted their [names matching strategy](https://github.com/Kotlin/kotlinx.serialization/blob/8723076c60baf11ef151c6a267e208f36024eb8b/formats/json/commonMain/src/kotlinx/serialization/json/internal/JsonNamesMap.kt#L85) and [schema cache](https://github.com/Kotlin/kotlinx.serialization/blob/master/formats/json/commonMain/src/kotlinx/serialization/json/internal/SchemaCache.kt) with lowercase encoding of enums.